packages/channel-google/src/channel-google.ts

```diff
@@ -0,0 +1,657 @@
+          externalId: c.adGroupAd.ad.id,
+          externalAdSetId: String(c.adGroup.id),
+        }));
+
+        const creatives: ChannelCreative[] = [];
+
+        const uniqueExternalIds = [...new Set(youtubeData.map((c) => c.adGroupAd.ad.id))];
+
+        const adsWithoutCreatives = await prisma.ad.findMany({
+          where: {
+            externalId: { in: uniqueExternalIds },
+            creativeId: null,
+          },
+          select: {
+            externalId: true,
+          },
+        });
+
+        for (const externalId of adsWithoutCreatives) {
+          const el = youtubeData.find((c) => c.adGroupAd.ad.id === externalId.externalId);
+
+          if (!el) continue;
+
+          if (el.adGroupAd.ad.videoResponsiveAd !== undefined) {
+            const query = `
+                  SELECT
+                  asset.id,
+                  asset.resource_name,
+                  asset.youtube_video_asset.youtube_video_id
+                FROM
+                  asset
+                WHERE
+                  asset.resource_name IN (${el.adGroupAd.ad.videoResponsiveAd.videos.map((c) => `'${c.asset}'`).join(',')})
+              `;
+
+            const asset = await Google.handlePagination(integration, query, assetResponseSchema, dbAccount.externalId);
+
+            if (!asset || isAError(asset)) return;
+
+            const flattenedCreatives: ChannelCreative[] = asset.map((c) => ({
+              externalAdId: el.adGroupAd.ad.id,
+              externalId: c.asset.id,
+              adAccountId: dbAccount.id,
+              name: c.asset.resourceName,
+              body: el.adGroupAd.ad.videoResponsiveAd?.descriptions?.[0].text,
+              title: el.video.title,
+              callToActionType: el.adGroupAd.ad.videoResponsiveAd?.callToActions?.[0].text,
+              imageUrl: c.asset.youtubeVideoAsset?.youtubeVideoId,
+            }));
+
+            creatives.push(...flattenedCreatives);
+          }
+        }
+
+        const insights: ChannelInsight[] = youtubeData.map((el) => ({
+          clicks: Number(el.metrics.clicks),
+          impressions: Number(el.metrics.impressions),
+          spend: Number(el.metrics.costMicros) / 1000,
+          externalAdId: el.adGroupAd.ad.id,
+          date: new Date(el.segments.date),
+          externalAccountId: dbAccount.externalId,
+          device: el.segments.device
+            ? (Google.deviceEnumMap.get(el.segments.device) ?? DeviceEnum.Unknown)
+            : DeviceEnum.Unknown,
+          publisher: PublisherEnum.Google,
+          position: 'feed',
+        }));
+
+        const uniqueInsights = Array.from(
+          new Map(
+            insights.map((insight) => [
+              `${insight.externalAdId}-${insight.date.toISOString()}-${insight.device}-${insight.publisher}-${insight.position}`,
+              insight,
+            ]),
+          ).values(),
+        );
+
+        await saveInsightsAdsAdsSetsCampaigns(
```

Comment:
```suggestion
        await deleteOldInsights(dbAccount.id, range.since, range.until);
        await saveInsightsAdsAdsSetsCampaigns(
```

This will solve the issue where the second time that you sync, sync fails.